hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+            )
+            gas_fee = NaN 
+            
+            
+        percent = self._maker_trade_fee if order_type is OrderType.LIMIT else self._taker_trade_fee 
+        
+        if order_type is OrderType.LIMIT:
+            return TradeFee(percent) #maker
+        
+        else:
+            return TradeFee(percent, flat_fees = [(quote_currency, gas_fee)]) #taker 
+    
+    
+    
+    async def calculateGasFees(self, quoteToken):
+        
+        fee_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/info"
+        
+        res = await self._api_request(http_method="get", url=fee_url)
+        
+        
+        feeData = res["data"]["base_spot_trading_fee_amounts"] #find by quoteToken
+        
+        baseAmount = float(feeData[quoteToken]["amount"]) / math.pow(10, 18)
+        
+        maxMatches = res["data"]["max_number_of_taker_matches_per_order"]
+        
+        takerGasFees = baseAmount * maxMatches
+        
+        
+
+    async def build_unsigned_order(self, amount: str, price: str, side: str, symbol: str, order_type: OrderType,
+                                   expires: int) -> Dict[str, Any]:
+        
+        
+        url = "%s/v1/orders/prepare" % (self.DOLOMITE_REST_ENDPOINT,)
+        
+        newAccount = web3.eth.account.create() #new auth address / private key 
+        
+        
+        baseToken = symbol.split('-')[0]
+        quoteToken = symbol.split('-')[1]
+        
+        secondaryAmount = 0
+        
+        if order_type is OrderType.MARKET and side == "buy":
+            secondaryAmount = 1000000000000 * math.pow(10, 18)
+            
+        elif order_type is OrderType.MARKET and side == "sell":
+            secondaryAmount = 1e-18 * math.pow(10, 18) #1  
+        else:
+            secondaryAmount = int((float(amount) * float(price)) * math.pow(10, 18))
+            
+            
+            
+        #Set fee amount for takers 
+        
+        fee_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/info"
+        
+        
+        res = await self._api_request(http_method="get", url=fee_url)
+        
+        
+        #Gas fees 
+        
+        feeData = res["data"]["base_spot_trading_fee_amounts"] #find by quoteToken
+        
+        baseAmount = float(feeData[quoteToken]["amount"]) / math.pow(10, 18)
+        maxMatches = res["data"]["max_number_of_taker_matches_per_order"]
+        
+        takerGasFees = baseAmount * maxMatches
+        
+        
+        #Commission fees
+        
+        market_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/markets/{symbol}"
+        
+        res2 = await self._api_request(http_method="get", url=market_url)
+        
+        marketPrice = float(res2["data"]["current_price"]["amount"]) / math.pow(10, 18)
+        
+        takerRate = float(res["data"]["taker_fee_percentage"])
+        
+        commissionAmount = marketPrice * takerRate * float(amount)
```

Comment:
Check if marketPrice is 0. If it is, replace it with the price of the order

Reply:
This is the only thing I haven't been able to address so far, because the "price" parameter passed into this method by the bot is essentially a null value in the case of a market order, so I wouldn't be able to set marketPrice = price (of order) in this case. Is there something else I can do to resolve this?